src/Portal.tsx

```diff
@@ -0,0 +1,34 @@
+import { PureComponent } from 'react';
+import * as ReactDOM from 'react-dom';
+
+export interface IPortalProps {
+  children: JSX.Element | string | number;
+  className?: string;
+}
+
+export class Portal extends PureComponent<IPortalProps> {
+  private readonly el!: HTMLDivElement;
+  private readonly root = document.body;
+
+  constructor(props: IPortalProps) {
+    super(props);
+
+    this.el = document.createElement('div');
```

Comment:
While not a strict requirement right now, I'd generally suggest protecting browser global usage in react components (for SSR). Something as simple as 

```js
if (typeof document === 'undefined') return;
```

Of course that means we should update `private readonly el!: HTMLDivElement;` to `private readonly el?: HTMLDivElement;` and add `if (!this.el) return null;` where appropriate.